### DISCUSSION TITLE
Help: How to compile SObjectizer with MingW on Linux Host Correctly?

### DISCUSSION THREAD

**Discussion by vidcentum** [Stiffstream/sobjectizer#86] [Q&A] [answered]
Hi,

May I request your help in  compiling SObjectizer with MingW on Linux Host correctly?
I am able to compile the library.
Host: Ubuntu 24.04
Toolchain: MingW
Build: CMake

I am able to compile the library. 

When I ran a sample program, the program got the following exception.

```
The procedure entry point __gthr_win32_self could not be located in the Dynamic link library.
```

Can anyone help me please?

Best Regards
Maruthi

**Comment by eao197** [comment]
Hi!

I've never used MinGW on Linux and don't understand what is going on.

Do you try to cross-compile the library on Linux and then run a sample on Windows?

**Reply by vidcentum** [reply]
Hello @eao197 :

Greetings.

Thank you for your quick reply. 

Yes, you are correct. I am trying to compile using MingW on Linux and run on Windows. 

`SObjectizer` is our favorite library for all our projects. So, we want to use it in the above scenario as well.

**We got this workaround:**
1. We pulled the sources of so_5 into the application source tree.
2. Added these  SO_5_PRJ,  SO_5__PLATFORM_REQUIRES_CDECL preprocessor directives in the build environment.
3. Now we could compile the so_5 and able to run on Windows and also with "wine"  on Linux itself for dev testing.
4. Our application is not time critical. 

The above Preprocessor directives are taken from CMakeLists.txt file in the so_5 library.

**Observation:**
It seems while building the library with CMake, the `if(WIN32) ...` in CMakeLists.txt  might not have been called.
```
if(WIN32)
	list(APPEND SO_5_DEFS "-DSO_5__PLATFORM_REQUIRES_CDECL")
endif()
```
May be passing -DWIN32=1 to CMake while building the library would do. I will check this and post the observation. 

I understand compiling along with app is not a correct option. For now, we could overcome the issue. And we will attempt to compile as library.

Appreciate your suggestions. Thanks once again.

Best Regards
Maruthi

**Reply by eao197** [reply, marked as answer]
It's pretty fine to place SObjectizer's sources into the project tree and compile them will all other project sources. We use such approach regularry.

Unfortunatelly, I have no experience with cross-compiling, so I have no any advices :(

It looks like MinGW specific problem. Maybe there is a difference between MinGW-provided libraries on Linux and Windows? Maybe versions of MinGW are different on both platforms?

**Reply by vidcentum** [reply]
@eao197 Thank you for your reply. 

Best Regards
Maruthi
